hopsworks-IT/src/test/ruby/spec/dataset_spec.rb

```diff
@@ -245,6 +245,70 @@
         end
       end
     end
+    describe "#upload" do
+      context 'without authentication' do
+        before :all do
+          with_valid_project
+          reset_session
+        end
+        it "should fail to upload" do
+          project = get_project
+          uploadFile(project, "Logs", "#{ENV['PROJECT_DIR']}/tools/metadata_designer/Sample.json")
+          expect_json(errorCode: 200003)
+          expect_status(401)
+        end
+      end
+      context 'with authentication but insufficient privilege' do
+        before :all do
+          with_valid_project
+        end
+        it "should fail to upload to a dataset with permission owner only" do
```

Comment:
@ErmiasG we can add `and role Data Scientist`. Data Owners are allowed from the REST API.

Reply:
maybe we can add a test for data owners as well?